### PR TITLE
Feature/Custom code: Core/App: Change - Unable to partial extend app. ED-514.

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/app/assets/js/ui/modal/modal.js [hunk 1]
```diff
@@ -42,9 +42,14 @@ ModalProvider.propTypes = {
 	children: PropTypes.node.isRequired,
 	toggleButtonProps: PropTypes.object.isRequired,
 	title: PropTypes.string.isRequired,
+	icon: PropTypes.string,
 };
 
-class Modal extends React.Component {
+ModalProvider.defaultProps = {
```
Comment (matipojo) [inline core/app/assets/js/ui/modal/modal.js:48]: Remove and use this icon from where the ModalProvider is rendered.

File: core/app/assets/js/ui/modal/modal.js [hunk 2]
```diff
@@ -98,4 +103,5 @@ Modal.propTypes = {
 	modalProps: PropTypes.object.isRequired,
 	children: PropTypes.any.isRequired,
 	title: PropTypes.string.isRequired,
+	icon: PropTypes.string.isRequired,
```
Comment (matipojo) [inline core/app/assets/js/ui/modal/modal.js:106]: Remove. it should be optional.

File: assets/dev/scss/direction/app-base.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+$direction: ltr;
```
Comment (matipojo) [inline assets/dev/scss/direction/app-base.scss:1]: Add an RTL version like other files in this folder.

File: assets/dev/scss/direction/app-base.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+@import "../../../../core/app/assets/styles/_functions.scss";
+@import "../../../../core/app/assets/styles/tokens.scss";
+@import "../../../../core/app/assets/styles/mixins.scss";
+@import "../../../../core/app/assets/js/ui/grid/grid";
+
+@import "../../../../core/app/assets/styles/base/utilities";
+@import "../../../../core/app/assets/styles/base/animations";
+@import "../../../../core/app/assets/js/ui/molecules/buttons";
+@import "../../../../core/app/assets/js/ui/modal/modal";
```
Comment (matipojo) [inline assets/dev/scss/direction/app-base.scss:13]: Extract to a file that imported here and in the full app file.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/app/app.php [hunk 5]
```diff
@@ -219,6 +219,8 @@ public function enqueue_app_loader() {
 		);
 
 		$this->print_config( 'elementor-app-loader' );
+
```
Comment (matipojo) [inline core/app/app.php:222]: Remove.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/app/assets/js/ui/modal/modal.js (hunk 1 above)
Comment (iNewLegend) [reply core/app/assets/js/ui/modal/modal.js]: Handled.

File: core/app/assets/js/ui/modal/modal.js (hunk 2 above)
Comment (iNewLegend) [reply core/app/assets/js/ui/modal/modal.js:102]: Handled.

File: assets/dev/scss/direction/app-base.scss (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/scss/direction/app-base.scss:1]: Handled.

File: core/app/app.php (hunk 5 above)
Comment (iNewLegend) [reply core/app/app.php]: Handled.

File: assets/dev/scss/direction/app-base.scss (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/scss/direction/app-base.scss]: Handled.

**Pull request closed without merging**